mm2src/mm2_main/src/rpc/lp_commands/tokens.rs

```diff
@@ -0,0 +1,95 @@
+use coins::eth::erc20::{get_erc20_ticker_by_contract_address, get_erc20_token_info, Erc20TokenInfo};
+use coins::eth::valid_addr_from_str;
+use coins::{lp_coinfind_or_err, CoinFindError, CoinProtocol, MmCoinEnum};
+use common::HttpStatusCode;
+use http::StatusCode;
+use mm2_core::mm_ctx::MmArc;
+use mm2_err_handle::prelude::*;
+
+#[derive(Deserialize)]
+pub struct TokenInfoRequest {
+    protocol: CoinProtocol,
+}
+
+#[derive(Serialize)]
+#[serde(tag = "type", content = "info")]
+pub enum TokenInfo {
+    ERC20(Erc20TokenInfo),
+}
+
+#[derive(Serialize)]
+pub struct TokenInfoResponse {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    config_ticker: Option<String>,
+    #[serde(flatten)]
+    info: TokenInfo,
+}
+
+#[derive(Debug, Deserialize, Display, Serialize, SerializeErrorType)]
+#[serde(tag = "error_type", content = "error_data")]
+pub enum TokenInfoError {
+    #[display(fmt = "No such coin {}", coin)]
+    NoSuchCoin { coin: String },
+    #[display(fmt = "Custom tokens are not supported for {} protocol yet!", protocol)]
+    UnsupportedTokenProtocol { protocol: String },
+    #[display(fmt = "Invalid request {}", _0)]
+    InvalidRequest(String),
+    #[display(fmt = "Error retrieving token info {}", _0)]
+    RetrieveInfoError(String),
+}
+
+impl HttpStatusCode for TokenInfoError {
+    fn status_code(&self) -> StatusCode {
+        match self {
+            TokenInfoError::NoSuchCoin { .. } => StatusCode::NOT_FOUND,
+            TokenInfoError::UnsupportedTokenProtocol { .. } | TokenInfoError::InvalidRequest(_) => {
+                StatusCode::BAD_REQUEST
+            },
+            TokenInfoError::RetrieveInfoError(_) => StatusCode::INTERNAL_SERVER_ERROR,
+        }
+    }
+}
+
+impl From<CoinFindError> for TokenInfoError {
+    fn from(e: CoinFindError) -> Self {
+        match e {
+            CoinFindError::NoSuchCoin { coin } => TokenInfoError::NoSuchCoin { coin },
+        }
+    }
+}
+
+pub async fn get_token_info(ctx: MmArc, req: TokenInfoRequest) -> MmResult<TokenInfoResponse, TokenInfoError> {
```

Comment:
I think, ideally this function could return the same struct that was used for token activation, like 'CoinConf' struct (or we could make a dedicated 'TokenConf' for tokens)

Reply:
Can you elaborate on how this `CoinConf`/`TokenConf` struct should look? btw, I did this based on the GUI mockups where the first screen needs this info to be auto populated similar to other wallets.

Reply:
Currently we parse the coins file as json.
I think, we could deserialise it to CoinConf and TokenConf (because coins and tokens configs are different)
Then we could use the same TokenConf for custom token activation and also return it as get_token_info result

Reply:
Yes, I guess this is for refactor (maybe we could do this backward compatible).
Plus it's good to add to refactor TODOs elimination of old-style eth_coin_from_conf_and_request as it requires to have settings in the token config which normally should be in platform coin (or moving it to for-test code part at least)